kliff/transforms/configuration_transforms/graphs/generate_graph.py

```diff
@@ -0,0 +1,199 @@
+from loguru import logger
+from monty.dev import requires
+
+from kliff.utils import torch_available, torch_geometric_available
+
+if torch_available():
+    import torch
+
+from typing import TYPE_CHECKING
+
+import kliff.transforms.configuration_transforms.graphs.graph_module as graph_module
+
+from ..configuration_transform import ConfigurationTransform
+
+if TYPE_CHECKING:
+    from kliff.dataset import Configuration
+
+if torch_geometric_available():
+    from torch_geometric.data import Data
+
+
+@requires(torch_geometric_available(), "torch_geometric is needed for KLIFFTorchGraph")
+class KLIFFTorchGraph(Data):
+    """
+    A Pytorch Geometric compatible graph representation of a configuration. When loaded
+    into a class:`torch_geometric.data.DataLoader` the graphs of type KLIFFTorchGraph
+    will be automatically collated and batched.
+
+    """
+
+    def __init__(self):
+        super(KLIFFTorchGraph, self).__init__()
+        self.num_nodes = (
+            None  # Simplify sizes and frees up pos key word, coords is cleaner
+        )
+        self.energy = None
+        self.forces = None
+        self.n_layers = None
```

Comment:
The major issue that arises in converting GNN to KIM-API is the graph updates. Most, if not all, GNNs uses a flavor of minimum image convention kind of node attribute list, where you only need graphs of atoms upto cutoff distance. In KIM-API we do not et global lattice information to do so. Hence graphs in KIM-API has to be reconstructed from purely cutoff/influence distance information. Therefore what this class do is that it calculates these graphs as convolution stages. That it if you want a 3 layer conv graph, it will generate graphs such that `infl_dist = cutoff * n_layers` and you get `edge_index0`, `edge_index1`, and `edge_index2` graphs where you can now apply the convolution operations in sequential fashion
```
h = conv(edge_index0, h)
h = conv(edge_index1, h)
h = conv(edge_index2, h)
```
as opposed to more conventional
```
for i in range(3):
   h = conv(edge_index,h)
```

This does have a benefit of making these operations parallel, as you no longer need global information. But they are essential for making graphs compatible to KIM-API.
Hence This class aggregates all the information needed for the graphs, including `n_layers`

Hope this helps.

Reply:
I would need to think a bit more on this to comment on if they are subset or not. I think they are, but need to confirm.